state/state.go

```diff
@@ -33,9 +36,9 @@ func NewClientWithOptions(ctx context.Context, conn *grpc.ClientConn, tableName
 	}
 	if slices.Contains(versions.Versions, 3) {
 		if opts.Versioned {
-			return stateV3.NewClientWithTable(ctx, pbPluginV3.NewPluginClient(conn), stateV3.VersionedTable(tableName))
+			return stateV3.NewClientWithTable(ctx, pbPluginV3.NewPluginClient(conn), stateV3.VersionedTable(tableName), conn)
```

Comment:
```suggestion
			return stateV3.NewClientWithTable(ctx, conn, stateV3.VersionedTable(tableName))
```

Maybe and then do `pbPluginV3.NewPluginClient(conn)` inside `NewClientWithTable`? Same comment a bit below

Reply:
Done in 9037389